src/main/java/baritone/pathing/movement/MovementHelper.java

```diff
@@ -280,6 +280,28 @@ static boolean avoidWalkingInto(Block block) {
                 || block == Blocks.WEB;
     }
 
+    static boolean isOverMagma(IPlayerContext ctx, BetterBlockPos src, BetterBlockPos dest) {
+        int x = ctx.playerFeet().x;
+        int y = ctx.playerFeet().y;
+        int z = ctx.playerFeet().z;
+        BlockStateInterface bsi = new BlockStateInterface(ctx);
+
+        //todo make this readable
+        boolean result = isOverMagma(bsi, x, y, z) ||                           //we are standing on a magma block
+                         isOverMagma(bsi, dest.x, y, dest.z) ||                 //the block we are going too, at the same y level is magma
+                         (isOverMagma(bsi, dest.x, dest.y, dest.z)              //the block we are going too is magma and
+                                 && y < src.y) ||                               //we are already dropping
+                         ((src.x != dest.x && src.z != dest.z) &&               //when moving diagonaly and
+                                         (isOverMagma(bsi, src.x, y, dest.z) || //we are cutting over magma or
+                                          isOverMagma(bsi, dest.x, y, src.z) || //we are cutting over magma or
+                                          isOverMagma(bsi, dest.x, y, dest.z))); //we are ending up on magma
+
+        return result;
+    }
```

Comment:
If I understand this correctly this intends to check whether the player is touching magma *right now*, so why not use the same kind of check as in `MovementDiagonal.isSafeToCancel`?